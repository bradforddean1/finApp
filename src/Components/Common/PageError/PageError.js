import React from "react";
import PropTypes from "prop-types";
import background from "../../../assets/elipse.svg";
import "./PageError.scss";

/**
 * Full Page Error
 * @component
 */
function PageError(props) {
	const { title, content } = props;

	return (
		<div
			className="PageError"
			style={{ backgroundImage: `url(${background})` }}
		>
			<h1>{title}</h1>
			<p>{content}</p>
		</div>
	);
}

PageError.propTypes = {
	title: PropTypes.string.isRequired,
	content: PropTypes.string,
};

PageError.defaultProps = {
	content: null,
};

export default PageError;
