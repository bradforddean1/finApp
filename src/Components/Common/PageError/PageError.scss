@use "../../../scss/variables" as v;

.PageError {
	height: 100%;
	display: flex;
	flex-direction: column;
	justify-content: center;

	font-family: v.$primary-font;
	text-align: center;
	padding: v.$gutter;

	background-position: center;
	background-size: cover;
	background-repeat: no-repeat;
}

.PageError h1 {
	margin-bottom: 40px;
	line-height: 60px;
	font-size: 60px;
}

.PageError p {
	font-size: 18px;
}
