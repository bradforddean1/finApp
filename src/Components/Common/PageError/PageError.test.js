import React from "react";
import ReactDOM from "react-dom";
import PageError from "./PageError";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("PageError Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<PageError />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<PageError />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	const props = { title: "Test Title", content: "Test Content" };

	it("Renders given props", () => {
		const wrapper = shallow(<PageError {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
