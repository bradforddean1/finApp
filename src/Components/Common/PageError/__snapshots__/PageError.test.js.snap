// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageError Component Renders given props 1`] = `
<div
  className="PageError"
  style={
    Object {
      "backgroundImage": "url(elipse.svg)",
    }
  }
>
  <h1>
    Test Title
  </h1>
  <p>
    Test Content
  </p>
</div>
`;

exports[`PageError Component Renders the Default State 1`] = `
<div
  className="PageError"
  style={
    Object {
      "backgroundImage": "url(elipse.svg)",
    }
  }
>
  <h1 />
  <p />
</div>
`;
